custom_components/tuya_local/devices/kogan_lx10_vacuum.yaml

```diff
@@ -83,9 +87,16 @@ primary_entity:
       type: boolean
     - id: 122
       name: error # robot_fault
-      # TODO: correctly parse the bitfield
-      type: integer
-      # "values": "{\"label\":[\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"23\",\"24\",\"25\",\"26\",\"27\",\"28\",\"29\",\"30\"],\"maxlen\":30}"
+      type: bitfield
+      mapping:
+        - dps_val: 0
+          value: 0
+        - dps_val: 32
+          value: Drop sensor abnormal
+        - dps_val: 524288
+          value: Machine trapped
+        - dps_val: 1048576
+          value: Was picked up at work
```

Comment:
Is this a description of the error, or your note?

Reply:
Those values are the messages displayed by the Tuya Smart app when the robot flags the given dps_val - this seemed the most sensible value to use, rather than attempting to distill it down to a single word or something else myself.